hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/CoprocessorHost.java

```diff
@@ -144,21 +144,44 @@ protected void loadSystemCoprocessors(Configuration conf, String confKey) {
 
     int currentSystemPriority = Coprocessor.PRIORITY_SYSTEM;
     for (String className : defaultCPClasses) {
-      String[] classNameAndPriority = className.split("\\|");
+      // After HBASE-23710 and HBASE-26714 when configuring for system coprocessor, we accept
+      // an optional format of className|priority|path
+      String[] classNameToken = className.split("\\|");
       boolean hasPriorityOverride = false;
-      className = classNameAndPriority[0];
+      boolean hasPath = false;
+      className = classNameToken[0];
       int overridePriority = Coprocessor.PRIORITY_SYSTEM;
-      if (classNameAndPriority.length > 1){
-        overridePriority = Integer.parseInt(classNameAndPriority[1]);
+      Path path = null;
+      if (classNameToken.length == 2) {
+        overridePriority = Integer.parseInt(classNameToken[1]);
         hasPriorityOverride = true;
       }
+      if (classNameToken.length == 3) {
```

Comment:
Shouldn't we set `overridePriority` here as well? Or is it supposed to be ignored and user `SYSTEM`? And what if the config definition is invalid, say `CLASS_NAME|PATH`? Should we perform some validation?

Reply:
I missed that and will fix , thanks for pointing that out